src/Evolution/Systems/CurvedScalarWave/Worldtube/Triggers/OrbitRadius.cpp

```diff
@@ -0,0 +1,47 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/CurvedScalarWave/Worldtube/Triggers/OrbitRadius.hpp"
+
+#include <array>
+#include <vector>
+
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp"
+#include "Time/Time.hpp"
+#include "Utilities/Serialization/CharmPupable.hpp"
+
+namespace Triggers {
+
+OrbitRadius::OrbitRadius(const std::vector<double>& radii) : radii_(radii) {}
+
+bool OrbitRadius::operator()(
+    const std::array<tnsr::I<double, 3, Frame::Inertial>, 2>&
+        position_and_velocity,
+    const TimeDelta& time_step) const {
+  const auto& position = position_and_velocity[0];
+  const auto& velocity = position_and_velocity[1];
+  const double current_radius = get(magnitude(position));
+  const double radial_velocity = (get<0>(position) * get<0>(velocity) +
+                                  get<1>(position) * get<1>(velocity) +
+                                  get<2>(position) * get<2>(velocity)) /
+                                 current_radius;
+  // factor 1.2 is for safety because the approximation is just linear
+  // approximation and triggering it multiple times is not a problem
+  const double next_radius =
+      current_radius + 1.2 * radial_velocity * time_step.value();
```

Comment:
Should 1.2 be an input option? Will it ever need to be changed?

Reply:
I do not think so. Eventually, I would really like to use an automated time step chooser but did not have any luck with them. I found that 1.2 will always trigger and rarely trigger twice which is completely fine for adjusting the time step.

This is not ideal but it is probably not worth finding a better solution for the scalar case at this point